### ISSUE TITLE
Tutorial issues list and questions for 3.0

### ISSUE THREAD

**Issue by losingkeys** [donejs/donejs#1128]
## Description

Typos, potential issues or confusion, and questions I had when going through the tutorial. I'm opening this early so we can work in parallel, but will fix remaining typos and straightforward issues myself when I've finished going through [the 3.0 tutorial](https://donejs.com/place-my-order.html).

## Possible issues

1. Re-loading a specific restaurant page with cmd+r gives the following error:
```
Failed promise:
XMLHttpRequest { getAllResponseHeaders: getAllResponseHeaders()
, onreadystatechange: onreadystatechange()
, readyState: 1, timeout: 0, withCredentials: false, upload: XMLHttpRequestUpload, responseURL: "", status: 0, statusText: "", responseType: "", … }
can-log.js:98:3
Potentially unhandled rejection [4] [object XMLHttpRequest] (WARNING: non-Error used) steal.js:140:22
unhandledRejection/logError<
http://localhost:8080/node_modules/steal/steal.js:140:22
report
http://localhost:8080/node_modules/steal/steal.js:167:5
flush
http://localhost:8080/node_modules/steal/steal.js:189:5
```
2. Deleted orders show up as blank spots in the list for me (is this a `null` thing?) rather than being removed from the list completely. Here's a screenshot (note, I have a browser plugin that makes it dark, sorry it's different):
<img width="833" alt="screen shot 2018-11-12 at 7 25 22 pm" src="https://user-images.githubusercontent.com/666493/48388415-2bbb4480-e6f1-11e8-95c9-2ebf98a6eeb8.png">
3. My images are gone after a prod build, and I'm pretty sure I've changed the build file properly and run the build commands correctly. Here's what it looks like:
<img width="364" alt="screen shot 2018-11-12 at 7 37 43 pm" src="https://user-images.githubusercontent.com/666493/48388568-b7cd6c00-e6f1-11e8-83ee-59d8c3bbde29.png">
4. I'm not sure if my cordova build worked... Here's the output:

```
Error: No platforms added to this project. Please use `cordova platform add <platform>`.


Done-> cordova build --release --buildFlag='-UseModernBuildSystem=0'
Error-> with Series tasks1
Using cordova CLI version (7.1.0)
Setting Current Working Directory (CWD) to ./build/cordova
Running:/Users/josh/projects/scratch/guides/place-my-order/node_modules/cordova/bin/cordova emulate ios --buildFlag='-UseModernBuildSystem=0'
Adding cordova-plugin-transport-security project...


Unable to load PlatformApi from platform. Error: Cannot find module '/Users/josh/projects/scratch/guides/place-my-order/build/cordova/node_modules/cordova-plugin-transport-security'


Error: Unhandled error. (The platform "cordova-plugin-transport-security" does not appear to be a valid cordova platform. It is missing API.js. cordova-plugin-transport-security not supported.)


Done-> cordova platform add ios

You have been opted out of telemetry. To change this, run: cordova telemetry on.

Error: No platforms added to this project. Please use `cordova platform add <platform>`.


Done-> cordova emulate ios --buildFlag='-UseModernBuildSystem=0'
Error-> with Parallel tasks1



 scratch/guides/place-my-order master %
```
I need to look into this more.. but does the output look okay to you? 

## Wording/possibly confusing things

1. The [Creating a GH Account/Repo](https://donejs.com/place-my-order.html#creating-a-github-account-and-repository) section feels a little awkward to me, maybe reword it? Here's what I was thinking: 
  > First, sign up (link) or sign in (link) to GitHub. Next, ensure you're set up to use the command line (link). Finally, create an empty project we can push our demo code to; here's what that looks like (pic). 
2. [The "New Order" form](https://donejs.com/place-my-order.html#write-the-template-1) example walks through the code pretty well, but I still don't totally understand what the dot is in this code: `order.items.has(.)`. Is there a link to the docs or an easy way to explain it?
3. [This section](https://donejs.com/place-my-order.html#bundling-assets) doesn't mention `build.js` until after you need to find it. It's easy to find (by searching for `stealTools` or reading on), but it was a bit of confusion for me at first.
4. Justin explained some of why Can tags are global... which is why we need the `Pmo` prefix on things... but this is never mentioned. I'd appreciate some further reading on how the tags work and why (though I know this outside the scope of the tutorial), but think it's at least worth mentioning the fact the tags are global and that's why we use the abbreviated app name as a prefix for components.
5. I didn't understand the `resolve(null)` [here](https://donejs.github.io/next/place-my-order.html#implement-view-model-behavior) at first... but after playing w/the app and reading the testing section it's pretty clear... the city section is unset upon selecting a new state, which is pretty cool. Is it worthwhile to mention this before the testing section?
6. Should "progressively loads"  be "asynchronously loads" in [this section](https://donejs.github.io/next/place-my-order.html#switch-between-components)?
7. [This section](https://donejs.com/place-my-order.html#create-a-loading-indicator) doesn't explain why `leak-scope` is added. Why is it needed?

## Typos

1. The [Create Routes](https://donejs.com/place-my-order.html#create-routes) section doesn't show the generated code (it's missing the `env:` key), though it does show you the lines you need to add.
2. The "enum comparitor" doc link in [this section](https://donejs.com/place-my-order.html#add-the-order-model-tag) 404s.

## Non-issue questions/curiosities

1. I'm guessing the generated fixtures code (in [Create a Test](https://donejs.com/place-my-order.html#create-a-test)) can't know about which fields are in the API, which is why it says `description:` rather than `name:`, right?
2. Should `"any"` be preferred over `"*"`? Or the other way around? If so there are some nits that could be picked :)
3. I'm pretty new to stealjs. Why are some libraries wrapped? Do they all need wrappers? Is there some optional benefit you get with wrappers that you don't always need?
4. Is there a glossary somewhere?
5. Is there somewhere that talks about building APIs that interop well with `realtimeRestModel`? Or how to configure ORMs to do the same?
6. Should more of the generated code have JSDoc documentation when generated, like this: ` connectedCallback(element) {}`?
7. The generated code [here](https://donejs.github.io/next/place-my-order.html#implement-view-model-behavior) uses two local variables so observable are set up correctly, right? Put another way, there's some possible issue with `&&` short-circuiting and observables, right? If so, should there be a link in the docs? If not, could you link me to some more reading?
8. Why return `null` in the getter [here](https://donejs.github.io/next/place-my-order.html#implement-view-model-behavior)? Would `undefined` work the same way?
9. Why `user-scalable=no` [here](https://donejs.com/place-my-order.html#loading-assets)?

## Potential enhancements to the guide or its site

1. It might be helpful if the code samples work like diffs, showing removals as well as additions. Some diffs (e.g. [this one](https://donejs.github.io/next/place-my-order.html#implement-view-model-behavior)) also remove things. The guide usually mentions what's removed in a note, or it's clear from context, or if you get an error. 
2. It would be cool to see phone number/form validation in a future version! Though this might not be much canjs/donejs magic if you just use html5 tags.
3. If the site that hosts the guide used `replaceState`, reloading the page would get you back to the section you were previously browsing. As it stands, it jumps to the last clicked on section.
4. The issue template on this repo asks you to use labels, but not everyone (myself right now, and I would assume most contributors most of the time) has access to those. How should those people categorize their issue? They could suggest tags, and maintainers could triage as needed. Either way I think it's a good idea to update the issue template.

## Environment

| Software                | Version
| ------------------ | -------
| donejs -V               | 3.0.0-pre.0
| node -v                  | v10.13.0
| npm -v                   | 6.4.1
| Browser                 | Chrome 70.0.3538.77
| Operating system | OS X 10.14.1

**Comment by matthewp** [comment]
Thanks for the detailed report!

**Comment by losingkeys** [comment]
No problem! I'll try to expand any vague parts (the stuff near the end of the tutorial could use more investigation on my part) and make typo PRs as I have time. I'm on a different project today though. Ping me in slack if you have questions though :)

**Comment by matthewp** [comment]
@losingkeys I was able to fix the first issue you found. Thanks for that.  I haven't been able to recreate the others. Do you still have the code? If so can you push your latest to github so I can take a look?

**Comment by losingkeys** [comment]
Yes ([here](https://github.com/losingkeys/place-my-order/)), but I ended up making a couple of copies in hopes of fixing some of the issues. Let me know if you still can't reproduce the issues, and I'll push up some other versions.

**Comment by matthewp** [comment]
These were all taken care of.
